controllers/workspace/finalize.go

```diff
@@ -30,13 +30,17 @@ import (
 
 const (
 	storageCleanupFinalizer = "storage.controller.devfile.io"
+	// devworkspacePhaseTerminating represents a DevWorkspace that has been deleted but is waiting on a finalizer.
+	// TODO: Should be moved to devfile/api side.
+	devworkspacePhaseTerminating devworkspace.WorkspacePhase = "Terminating"
```

Comment:
I wrongly assumed that our DevWorkspace can wait dependent objects to be removed with empty finalizers.

But it's not true. According to https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/
```
The object's metadata.finalizers contains the value "foregroundDeletion".
```

So, if we want DevWorkspace to wait dependent objects, technically it's finalizing.
From that perspective, `"Finalizing"` phase a bit win, but just a bit since:
- `Terminating` can be confused with `Stopping`, but if you know K8s API and Namespaces are terminating before removing, it's pretty clear;
- `Finalizing` can be confused with anything, like finalizing after stop, finalizing after start. If you know about K8s garbage collector, you probably will guess to check DevWorkspace CR finalizers, to check the hanging cause, in case it's third-party finalizer.

I'm 50\50 between `Terminating` and `Finalizing`

Alternative from tmp-che-devworkspace:
`REMOVAL_WAIT`, `FINALIZING_DELETE`, `PendingDeletion`(DWO uses one when there are still workspaces on the cluster)

Reply:
from https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/
> When a Pod is being deleted, it is shown as Terminating by some kubectl commands. This Terminating status is not one of the Pod phases.

Reply:
for namespace it's 

> Terminating the namespace is being deleted, and can not be used for new objects

which may be for devworkspace:

> Terminating the devworkspace is being deleted, cleaning up the dependent resources, can't not be started anymore